docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md

```diff
@@ -1,139 +1,243 @@
-## This document lists known breaking changes in Roslyn after .NET 6 all the way to .NET 7.
+# This document lists known breaking changes in Roslyn after .NET 6 all the way to .NET 7.
 
-1. <a name="1"></a>In Visual Studio 17.1, the contextual keyword `var` cannot be used as an explicit lambda return type.
+## Foreach enumerator as a ref struct
```

Comment:
I also removed the hand-coded anchors. The H2s will have automatically generated anchors. 

I searched and didn't find any links to these anchors.

Reply:
There is a link in the [DotNet 6 document](https://github.com/dotnet/roslyn/blob/main/docs/compilers/CSharp/Compiler%20Breaking%20Changes%20-%20DotNet%206.md?plain=1#L147).

Reply:
added in the 3rd commit.